### PR TITLE
[SPARK-48490][CORE] Unescapes any literals for message of MessageWithContext

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to `unescapes` any literals for `message` of `MessageWithContext`

### Why are the changes needed?
- For example, before this PR
```
logInfo("This is a log message\nThis is a new line \t other msg")
```
It will output:
```
24/05/31 22:53:27 INFO PatternLoggingSuite: This is a log message
This is a new line 	 other msg
```

But:

```
logInfo(log"This is a log message\nThis is a new line \t other msg")
```
It will output:
```
24/05/31 22:53:59 ERROR PatternLoggingSuite: This is a log message\nThis is a new line \t other msg
```

Obviously, the latter is not the result we `expected`.

### Does this PR introduce _any_ user-facing change?
Yes, fix bug.

### How was this patch tested?
- Add new UT.
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @gengliangwang

File: common/utils/src/main/scala/org/apache/spark/internal/Logging.scala [hunk 1]
```diff
@@ -158,7 +159,7 @@ trait Logging {
         }
       }
 
-      MessageWithContext(sb.toString(), context)
+      MessageWithContext(StringEscapeUtils.unescapeJava(sb.toString()), context)
```
Comment (gengliangwang) [inline common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:162]: shall we change 
```
class LogEntry(messageWithContext: => MessageWithContext) {
  def message: String = messageWithContext.message
```
instead? 
There are many log concations. We should avoid calling the method `unescapeJava` for all the log fragments.

Comment (panbingkun) [reply common/utils/src/main/scala/org/apache/spark/internal/Logging.scala]: Updated.

**Comment by panbingkun** [comment]
@gengliangwang 
In addition, we also encountered this issue during migration: https://github.com/apache/spark/pull/46822

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

File: common/utils/src/main/scala/org/apache/spark/internal/Logging.scala [hunk 2]
```diff
@@ -99,7 +100,7 @@ case class MessageWithContext(message: String, context: java.util.HashMap[String
  * Companion class for lazy evaluation of the MessageWithContext instance.
  */
 class LogEntry(messageWithContext: => MessageWithContext) {
-  def message: String = messageWithContext.message
+  def message: String = StringEscapeUtils.unescapeJava(messageWithContext.message)
```
Comment (dongjoon-hyun) [inline common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:103]: Hi, @panbingkun and @gengliangwang .

This seems to break SparkR somehow. Could you take a look at this?

- https://github.com/apache/spark/actions/workflows/build_sparkr_window.yml

```
-- Error ('test_basic.R:25:3'): create DataFrame from list or data.frame -------
Error in `handleErrors(returnStatus, conn)`: java.lang.IllegalArgumentException: Unable to parse unicode value: serF
	at org.apache.commons.text.translate.UnicodeUnescaper.translate(UnicodeUnescaper.java:55)
	at org.apache.commons.text.translate.AggregateTranslator.translate(AggregateTranslator.java:58)
	at org.apache.commons.text.translate.CharSequenceTranslator.translate(CharSequenceTranslator.java:101)
	at org.apache.commons.text.translate.CharSequenceTranslator.translate(CharSequenceTranslator.java:63)
	at org.apache.commons.text.StringEscapeUtils.unescapeJava(StringEscapeUtils.java:802)
	at org.apache.spark.internal.LogEntry.message(Logging.scala:103)
```

Comment (panbingkun) [reply common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:103]: Okay, Let me investigate it

Comment (panbingkun) [reply common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:103]: @dongjoon-hyun
I have identified the root cause of this issue and am currently attempting to fix it
https://github.com/apache/spark/pull/46897#issuecomment-2177869329

**Pull request closed without merging**